src/name_operations.jl

```diff
@@ -39,9 +39,25 @@ unname(x::AbstractArray) = x
 
 """
     dimnames(A) -> Tuple
+    dimnames(A, d) -> Symbol
 
-Return the names of all the dimensions of the array `A`.
+Return the names of all the dimensions of the array `A`, 
+or just the one for the `d`-th dimension. 
+
+Gives wildcards `:_` if this is not a `NamedDimsArray`.
+And like `size(A, d)`, it allows `d > ndims(A)`.
 """
-dimnames(::Type{<:NamedDimsArray{L}}) where L = L
+dimnames(::Type{<:NamedDimsArray{L}}) where {L} = L
 dimnames(::Type{<:AbstractArray{T, N}}) where {T, N} = ntuple(_->:_, N)
 dimnames(x::T) where T<:AbstractArray = dimnames(T)
+
+function dimnames(AT::Type{<:AbstractArray{T,N}}, d::Integer) where {T,N}
+    if d in 1:N
+        return dimnames(AT)[d]
+    elseif d>N
+        return :_
+    else
+        error("dimnames: dimension out of range")
```

Comment:
I think we'd want this to be a BoundsError, like
```julia
julia> size(1:3, 0)
ERROR: BoundsError: attempt to access (3,)
  at index [0]
```

I think we can have that with 
```julia
function dimnames(AT::Type{<:AbstractArray{T,N}}, d::Integer) where {T,N}
    return d > N ? :_ : dimnames(AT)[d]
end
```
```julia
julia> dimnames([10 20; 30 40], 0)
ERROR: BoundsError: attempt to access (:_, :_)
  at index [0]
```

Reply:
I followed this behaviour:
```
julia> try size(ones(2),0) catch e e end
ErrorException("arraysize: dimension out of range")
```
but indeed it seems the Base isn't consistent about this.

Reply:
Fair enough. Silly Base 😄 I think plain `ErrorExceptions` are less helpful, and the BoundsError seems right here, so i'd prefer we go with that, if you're fine with it?

Otherwise I think this looks good 👍